src/controllers/auth.controller.js

```
@@ -0,0 +1,267 @@
+const User = require('../models/User');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const sendMail = require('../utils/sendMail');
+
+const JWT_SECRET = process.env.JWT_SECRET || 'supersecretkey';
```

You provide fallbacks for JWT secrets (`'supersecretkey'`, etc.). Hard-coded default secrets are insecure and can hide misconfiguration. Prefer failing fast if secrets are missing or centralize dotenv loading and config validation at startup so missing env vars are detected early.